### PR TITLE
Improve chat feature allowing farmers connect with experts in real time

### PR DESCRIPTION
**Pull request by trangttdang**
Farmers and experts now can send and receive messages in real time. For better chat feature, I will modify it using ParseLiveQuery Chat to decrease response time, high network and server usage

### CODE REVIEW

File: Farmacy/ViewController/ChatViewController.m
Comment (xmdou) [inline Farmacy/ViewController/ChatViewController.m]: Are these blank line on purpose? Please delete them before merge the code. Thanks.

File: Farmacy/ViewController/ChatViewController.m
Comment (xmdou) [inline Farmacy/ViewController/ChatViewController.m]: Better to create constant value for "User", then use the constant value for line 36, line 56, and line 71. Same with "text". Furthermore, could you please keep "User" and "text" the same format, either "User" and "Text", or "user" and "text"

File: Farmacy/ViewController/LoginViewController.m
Comment (xmdou) [inline Farmacy/ViewController/LoginViewController.m:38]: It's not a good idea to compare/save/display the user the password. You can encrypt it, some options https://en.wikipedia.org/wiki/Secure_Hash_Algorithms. But for intern project, you can keep it like this to unblock other features implementation. Please keep this as a long term action item. Thanks.

**Review by xmdou — APPROVED** [review]
accept to unblock features, please update the code in a further review. Thanks.

File: Farmacy/ViewController/LoginViewController.m
Comment (jianghaolu) [inline Farmacy/ViewController/LoginViewController.m:42]: nit: should we display an error if login fails?

File: Farmacy/Views/ConversationCell.h
Comment (jianghaolu) [inline Farmacy/Views/ConversationCell.h:16]:
```suggestion
@property (weak, nonatomic) id<ConversationCellDelegate> delegate;
```

**Review by jianghaolu — APPROVED** [review]
Overall lgtm. Feel free to merge and address in the next PR.

File: Farmacy/ViewController/LoginViewController.m
Comment (trangttdang) [reply Farmacy/ViewController/LoginViewController.m:42]: I've updated with Login Alert Message to alert users and allow them to log in again

**Pull request merged**
